Android/islandRush/app/src/main/java/com/quinstedt/islandRush/BrokerConnection.java

```diff
@@ -143,6 +222,66 @@ public void messageArrived(String topic, MqttMessage message) throws Exception {
+                            player5.setText(payload);
+                    }else if (topic.equals(firstPlayerName)) {
+                            String payload = new String(message.getPayload());
+                            firstPlayer.setText(payload);
+                    }else if (topic.equals(secondPlayerName)) {
+                            String payload = new String(message.getPayload());
+                            secondPlayer.setText(payload);
+                    }else if (topic.equals(thirdPlayerName)) {
+                            String payload = new String(message.getPayload());
+                            thirdPlayer.setText(payload);
+                    }else if (topic.equals(fourthPlayerName)) {
+                            String payload = new String(message.getPayload());
+                            fourPlayer.setText(payload);
+                    }else if (topic.equals(fifthPlayerName)) {
+                            String payload = new String(message.getPayload());
+                            fifthPlayer.setText(payload);
+                    }else if (topic.equals(time1)) {
+                            String text = new String(message.getPayload());
+                            firstTime.setText(text);
+                    }else if (topic.equals(time2)) {
+                            String text = new String(message.getPayload());
+                            secondTime.setText(text);
+                    }else if (topic.equals(time3)) {
+                            String text = new String(message.getPayload());
+                            thirdTime.setText(text);
+                    }else if (topic.equals(time4)) {
+                            String text = new String(message.getPayload());
+                            fourthTime.setText(text);
+                    }else if (topic.equals(time5)) {
+                            String text = new String(message.getPayload());
+                            fifthTime.setText(text);
+                    }else if (topic.equals(avgSpeed1)) {
+                            String text = new String(message.getPayload());
+                            speed1.setText(text);
+                    }else if (topic.equals(avgSpeed2)) {
+                            String text = new String(message.getPayload());
+                            speed2.setText(text);
+                    }else if (topic.equals(avgSpeed3)) {
+                            String text = new String(message.getPayload());
+                            speed3.setText(text);
+                    }else if (topic.equals(avgSpeed4)) {
+                            String text = new String(message.getPayload());
+                            speed4.setText(text);
+                    }else if (topic.equals(avgSpeed5)) {
+                            String text = new String(message.getPayload());
+                            speed5.setText(text);
```

Comment:
You seem to want to **map** strings to `TextView` elements.
So what you need instead is to create a `HashMap<String, TextView>` (read up on what it does if not familiar with it) and get rid of this unsustainable list of if statements.

Reply:
We were thinking about using Switch statements for better readability but I'll think about using hashmaps

Reply:
That will work too. Overall, every time you are copy-pasting code, chances are you are doing something wrong. :)

Reply:
We can look into a few solutions and see which one will be the most cost effective one to use.

1. Parsin JsonFiles: 

Send the hole Jsonfile as a String and Parsing it in the app to Json again.  
Then sending it back when the app is close to the app is closed.  

2. Merge the Server with the App.

The timer is now being implemented on in the App.
The server will now be storing the leaderboard and doing some calculation on the distance and averge speed. 
Which are topics that are all ready in the app aswell. But this will lead that the leaderboard is only for one app.

             2.1 Implemeting a database ( SQL and Firebase are good options) will give the app the multiplayer leaderboard 
                   property again

3. Use a Switch case for the topics or a HashMap. But if we are using the Hashmap then we need to have the TextView as the key and the give it the string from the MqttMessage as a value. 

@platisd what do you think?

Reply:
I am not super familiar with your code base, so I can't answer 1 & 2 properly. However, regarding JSON, I advise against sending it from the microcontroller since you will run into various technical difficulties and handling JSON on a machine with a very restrained heap memory is potentially tricky.

A HashMap will be the least lines of code (I think) and potentially one of the most readable solutions, so I'd say go for that.
It will eventually look something like `dashboard[topic].setText(new String(message.getPayload()));`.

Generally speaking and unrelated to the above, architecturally it's not the best idea to have your UI as tightly coupled with the communication layer as you do here, but whatever, that will work for now.

Reply:
@Quinstedt  I was thinking about working with option 2. Merging the server with the app, then we don't need all these mqtt connection and if statements. And as Dimitris said it's not the best idea to have the UI tightly coupled with the communication layer. I'll look into it and see if the merging works.